pkg/cloud/vsphere/context/machine_context.go

```diff
@@ -193,15 +195,15 @@ func (c *MachineContext) IPAddr() string {
 		}
 
 		if preferredAPIServerCIDR == nil {
-			return nodeAddr.Address
+			return nodeAddr.Address, nil
 		}
 
 		if preferredAPIServerCIDR.Contains(net.ParseIP(nodeAddr.Address)) {
-			return nodeAddr.Address
+			return nodeAddr.Address, nil
 		}
 	}
 
-	return ""
+	return "", ErrNoMachineIPAddr
```

Comment:
Technically don't think this error var is needed yet since any error from `IPAddr()` results in a requeue so we don't actually have to check the returned error value yet. With that said it does make the code path and unit tests easier to read so I kept it here